apps/member-profile/app/shared/queries/index.ts

```diff
@@ -208,3 +208,15 @@ export async function listEmails(id: string) {
 
   return rows;
 }
+
+export async function updateAllowEmailShare(
+  trx: Transaction<DB>,
+  id: string,
+  allowEmailShareBool: boolean
+) {
+  await trx
+    .updateTable('students')
+    .set({ allowEmailShare: allowEmailShareBool })
+    .where('id', '=', id)
+    .execute();
+}
```

Comment:
In an effort to colocate core backend logic, let's move this to the `core` package and create a use-case in the `member` module. You can find an example [here](https://github.com/colorstackorg/oyster/blob/main/packages/core/src/modules/member/use-cases/join-member-directory.ts).